db/record.c

```diff
@@ -265,25 +264,30 @@ int add_record(struct ireq *iq, void *trans, const uint8_t *p_buf_tag_name,
 
     if (!(flags & RECFLAGS_NEW_SCHEMA)) { // dont lock if adding from SC
 
-        int d_ms = BDB_ATTR_GET(thedb->bdb_attr, DELAY_LOCK_TABLE_RECORD_C);
+        int d_ms = BDB_ATTR_GET(thedb->bdb_attr, DELAY_WRITES_IN_RECORD_C);
         if (d_ms) {
             if (iq->debug)
-                reqprintf(iq, "Sleeping for %d ms", d_ms);
-            usleep(1000 * d_ms);
+                reqprintf(iq, "Sleeping for DELAY_WRITES_IN_RECORD_C (%dms)", d_ms);
+            int lrc = usleep(1000 * d_ms);
+            if (lrc)
+                reqprintf(iq, "usleep error rc %d errno %d\n", rc, errno);
         }
 
-        rc = bdb_lock_table_read(iq->usedb->handle, trans);
-        if (rc == BDBERR_DEADLOCK) {
-            if (iq->debug)
-                reqprintf(iq, "LOCK TABLE READ DEADLOCK");
-            retrc = RC_INTERNAL_RETRY;
-            ERR;
-        } else if (rc) {
-            if (iq->debug)
-                reqprintf(iq, "LOCK TABLE READ ERROR: %d", rc);
-            *opfailcode = OP_FAILED_INTERNAL;
-            retrc = ERR_INTERNAL;
-            ERR;
+        if(!iq->have_blkseq) {
```

Comment:
Why is have_blkseq relevant? How does `if (!iq->is_sorese)` sound?

Reply:
Hmm, that might work too, i didn't look. It looked like have_blkseq was set for sql and looked like what we wanted; sorese might do the same, i'll check.